src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1185,12 +1185,25 @@ def __init__(self, label=None, hosts=None, count=None):
 
 
         self.count = count  # type: Optional[int]
+        self.all_hosts = all_hosts  # type: bool
 
     def set_hosts(self, hosts):
         # To backpopulate the .hosts attribute when using labels or count
         # in the orchestrator backend.
         self.hosts = hosts
 
+    def __repr__(self):
```

Comment:
can be shortened with

```python
    def __repr__(self):
        return "{}({!r})".format(self.__class__.__name__, self.__dict__)

```

Reply:
That's simpler, but more verbose:
```
1: >       assert str(ret) == expected
1: E       assert "PlacementSpe...osts': True})" == 'PlacementSpec(all=true)'
1: E         - PlacementSpec({'label': None, 'hosts': [], 'count': None, 'all_hosts': True})
1: E         + PlacementSpec(all=true)
```
also i'm worried the key order is nondeterministic? I remember having to fix a bunch of tests when the dict key order changed between python versions.

(Also note this patch belongs to #33465 )